Signup: Update DesignTypeWithAtomicStoreStep to use TileGrid

tyxla opened the pull request:
This PR updates `DesignTypeWithAtomicStoreStep` to use the `TileGrid` component family for displaying all the tiles. It also reuses some classes from `DesignTypeWithStoreStep` - we can later move these to a more common location, so they can be reused. 

A change that's worth mentioning is that it will now rely on `this.state.showStore` directly for showing/hiding the atomic store, vs. unnecessarily using a className for that.

This allows to save a bunch of CSS and markup that we're reusing in other design type steps in Signup and Jetpack Onboarding. Can be followed by a couple more cleanup PRs after merging.

Contains #20640, which fixes a small visual bug with card link indicators on mobile devices.

To test:
* Checkout this branch
* Go to http://calypso.localhost:3000/start/design-type-with-store-nux
* Try both different variants of the `signupAtomicStoreVsPressable` AB test.
* Verify there are no visual/functional differences when comparing to production.
* Verify the step works like it did before.

MichaelArestad commented on client/components/tile-grid/style.scss:
```diff
@@ -90,6 +94,7 @@
 			border-top: 2px solid lighten( $gray, 20% );
 			border-right: 2px solid lighten( $gray, 20% );
 			transform: rotate(45deg);
+			color: #fff;
```
Can we make this `$white`

tyxla replied:
🤦‍♂️

tyxla replied:
Fixed in 767901dbaa7d15a800ddb58384cd9d08744af165.

MichaelArestad commented on client/signup/steps/design-type-with-atomic-store/style.scss:
```diff
@@ -1,193 +0,0 @@
-.design-type-with-atomic-store {
```
I love all this red

lamosty replied:
First we copy it and then we delete it 💯

tyxla commented:
This one is ready to go from my perspective; could anyone please test and code-review it? Thanks!

tyxla commented:
Good catch @sirreal! It seems there were visual issues showing the Pressable store.

However, I've reverted the approach a bit, because after fixing the visual issue that was preventing the Pressable store from appearing, the previous approach was erasing animations between steps. 

I've updated the PR, feel free to have another look and test it again. Thanks!

sirreal commented:
I tested this and it seems to work well now. I'll give it a thorough review and look the code over tomorrow.

sirreal commented on client/signup/steps/design-type-with-atomic-store/index.jsx:
```diff
@@ -12,15 +12,10 @@ import { includes, invoke } from 'lodash';
-	PageImage,
-	GridImage,
-	StoreImage,
-} from '../design-type-with-atomic-store/type-images';
```
Can we remove the entire file these are imported from?

https://github.com/Automattic/wp-calypso/blob/e9a8179b5a30238927d65c499fe5357018e99645/client/signup/steps/design-type-with-atomic-store/type-images.jsx#L1-L23

tyxla replied:
👍  8ce9473bd06dee6874984db661f55d1f60bab94f

sirreal commented on client/signup/steps/design-type-with-atomic-store/index.jsx:
```diff
@@ -187,9 +176,13 @@ class DesignTypeWithAtomicStoreStep extends Component {
 				</div>
 				<div className={ designTypeListClassName }>
-					{ this.getChoices().map( this.renderChoice ) }
+					<TileGrid className="design-type-with-atomic-store__list">
```
Look like this `className` can be removed.

tyxla replied:
👍  2eedfce893f6ca32fc4db13fb0f48121159b32c7

sirreal commented on client/signup/steps/design-type-with-atomic-store/index.jsx:
```diff
@@ -172,12 +161,12 @@ class DesignTypeWithAtomicStoreStep extends Component {
 
 		return (
-			<div className="design-type-with-atomic-store__substep-wrapper">
+			<div className="design-type-with-atomic-store__substep-wrapper design-type-with-store__substep-wrapper">
```
Both classNames are needed for namespace lint rule?

tyxla replied:
The first one is necessary for the lint rule, the second one - for making the container relative. Can be cleaned up after we migrate the rest of the steps to use `TileGrid`.

tyxla commented:
Thank you all for reviewing/testing 👍
